src/syncProvider.ts

```diff
@@ -128,7 +128,17 @@ export class FTPSyncProvider implements ISyncProvider {
         this.logger.all(`removing folder "${absoluteFolderPath}"`);
 
         if (this.dryRun === false) {
-            await retryRequest(this.logger, async () => await this.client.removeDir(absoluteFolderPath));
+            try {
+                await retryRequest(this.logger, async () => await this.client.removeDir(absoluteFolderPath));
+            }
+            catch (e: any) {
+                if (e.code === ErrorCode.FileNotFoundOrNoAccess) {
```

Comment:
e.code could throw (e is type any, code might not be defined)

Can you use `e?.code === ...` instead?